template.yml

```diff
@@ -821,7 +821,8 @@ Resources:
             Action: sts:AssumeRole
 
   STMAnalyzeSegments:
-    Type: AWS::Serverless::Function # More info about Function Resource: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction
+    Condition: STMAnalyzeSegmentsEnabled
```

Comment:
The condition should not appear anywhere else than the alias and version blocks otherwise your function will be erased or launch errors when your condition is False.